Merge greeterAsync into greeter (C++) (#256)
zeroc-ice/ice-demos: cpp/Ice/greeter/README.md

# Greeter

The Greeter demo illustrates how to build a simple application with Ice.

> The client demonstrates the 3 APIs you can use to make remove invocations using generated Ice proxies:

Reviewer: There's a typo, and saying "the client" sounds weird and ambiguous. We were just talking about 'the greeter' and didn't mention anything about a client until now.
```suggestion
It features a client which demonstrates the 3 APIs you can use to make remote invocations using generated Ice proxies:
```

- synchronous (`greet`),

> - asynchronous with future (`greetAsync`); and

Reviewer: ```suggestion
- asynchronous with futures (`greetAsync`); and
```
I feel like this should be plural? Same as we do for "callbacks" below.

- asynchronous with callbacks (another `greetAsync`)

> This demo alo provides two implementations for the server: a synchronous dispatch implementation (`server`), and an

Reviewer: ```suggestion
This demo also provides two implementations for the server: a synchronous dispatch implementation (`server`), and an
```

> asynchronous dispatch implementation (`serveramd`). The same client works with both.

Reviewer: I don't see why we say `same`. There's only one client in this demo right?
What is this client the "same" as?
```suggestion
asynchronous dispatch implementation (`serveramd`). The client works with both.
```

To build the demo, run:

```shell
cmake -B build
cmake --build build --config Release
```

> The build produces 3 executables: client, server and serveramd.

Reviewer: Ever the pedant:
```suggestion
The build produces 3 executables: client, server, and serveramd.
```
